experiments/integrated/fluxnet/run_fluxnet.jl

```diff
@@ -265,14 +270,82 @@ prob = SciMLBase.ODEProblem(
     (t0, tf),
     p,
 );
+
+# ClimaDiagnostics
+# ClimaDiagnostics
+using ClimaDiagnostics
+using ClimaUtilities
+
+outdir = joinpath(pkgdir(ClimaLand), "experiments/integrated/fluxnet/out")
+output_dir = ClimaUtilities.OutputPathGenerator.generate_output_path(outdir)
+
+d_writer = ClimaDiagnostics.Writers.DictWriter()
+
+ref_time = DateTime(2005) # random. not sure what it should be
+
+diags = ClimaLand.default_diagnostics(
+    land,
+    ref_time;
+    output_writer = d_writer,
+    output_vars = :long,
+    average_period = :hourly,
+)
+
+diagnostic_handler =
+    ClimaDiagnostics.DiagnosticsHandler(diags, Y, p, t0, dt = dt);
+
+diag_cb = ClimaDiagnostics.DiagnosticsCallback(diagnostic_handler);
+
+drivers_diag = ClimaLand.get_drivers(land)
+updatefunc = ClimaLand.make_update_drivers(drivers_diag)
+driver_cb = ClimaLand.DriverUpdateCallback(updateat, updatefunc)
+
 sol = SciMLBase.solve(
     prob,
     ode_algo;
     dt = dt,
-    callback = cb,
-    adaptive = false,
-    saveat = saveat,
-)
+    callback = SciMLBase.CallbackSet(driver_cb, diag_cb),
+);
+
+short_names_1D = [
+    "sif", # SIF
+    "ra", # AR
+    "gs", # g_stomata
+    "gpp", # GPP
+    "ct", # canopy_T
+    "swu", # SW_u
+    "lwu", # LW_u
+    "er", # ER
+    "et", # ET
+    "msf", # β
+    "shf", # SHF
+    "lhf", # LHF
+    "ghf", # G
+    "rn", # Rn
+]
+
+short_names_2D = [
+    "swc", # swc_sfc or swc_5 or swc_10
+    "tsoil", # soil_T_sfc or soil_T_5 or soil_T_10
+    "si", # si_sfc or si_5 or si_10
+]
+
+hourly_diag_name = short_names_1D .* "_1h_average"
+hourly_diag_name_2D = short_names_2D .* "_1h_average"
+
+SIF, AR, g_stomata, GPP, canopy_T, SW_u, LW_u, ER, ET, β, SHF, LHF, G, Rn = [
+    ClimaLand.Diagnostics.diagnostic_as_vectors(d_writer, diag_name, 1)[2][1:721]
+    for diag_name in hourly_diag_name
+] # [1:721] only because summer subset of model output
```

Comment:
This doesn't scale very well. `ClimaLand.Diagnostics.diagnostic_as_vectors` returns the time too, can you use that to  detect when it is summer instead?

Reply:
I agree that this old script had many hardcoded stuff that make it difficult to maintain, and should be refactored properly.

however for instance, we were doing summer only because there was no snow model. once we have snow, we will run the whole year/dataset. so this is temporary.